samanthas_telegram_bot/api_queries/send_get.py

```diff
@@ -0,0 +1,293 @@
+        parse_mode=None,
+        logger=logger,
+        level="error",
+    )
+    return 0
+
+
+async def send_student_info(update: Update, user_data: UserData) -> bool:
+    """Sends a POST request to create a student and send results of assessment if any."""
+
+    personal_info_id = await _send_personal_info_get_id(user_data=user_data, bot=update.get_bot())
+
+    data = {
+        "personal_info": personal_info_id,
+        "comment": user_data.comment,
+        "status": STATUS_AT_CREATION_STUDENT_TEACHER,
+        "status_since": _format_status_since(update),
+        "can_read_in_english": user_data.student_can_read_in_english,
+        "is_member_of_speaking_club": False,  # TODO can backend set to False by default?
+        "age_range": user_data.student_age_range_id,
+        "availability_slots": user_data.day_and_time_slot_ids,
+        "non_teaching_help_required": user_data.non_teaching_help_types,
+        "teaching_languages_and_levels": user_data.language_and_level_ids,
+    }
+
+    if user_data.student_smalltalk_result:
+        data["smalltalk_test_result"] = json.dumps(
+            str(user_data.student_smalltalk_result.original_json)
+        )
+        # TODO url (when field in backend is created)
+
+    async with httpx.AsyncClient() as client:
+        r = await client.post(f"{API_URL_PREFIX}/students/", data=data)
+
+    if r.status_code != httpx.codes.CREATED:
+        await log_and_report(
+            text=(
+                f"Chat {user_data.chat_id}: Failed to create student ({r.status_code=}, "
+                f"{r.content=})"
+            ),
+            logger=logger,
+            level="error",
+            bot=update.get_bot(),
+            parse_mode=None,
+        )
+        return False
+
+    logger.info(f"Chat {user_data.chat_id}: Created student ({personal_info_id=})")
+    await send_to_admin_group(
+        bot=update.get_bot(),
+        text=(
+            f"New student: [{user_data.first_name} {user_data.last_name}]"
+            f"({API_URL_PREFIX}/student/{personal_info_id})"
+        ),
+        parse_mode=ParseMode.MARKDOWN_V2,
+    )
+
+    if not user_data.student_assessment_answers:
+        logger.info(f"Chat {user_data.chat_id}: no assessment answers to send")
+        return True
+
+    async with httpx.AsyncClient() as client:
+        r = await client.post(
+            f"{API_URL_PREFIX}/enrollment_test_result/",
+            data={
+                "student": personal_info_id,
+                "answers": [item.answer_id for item in user_data.student_assessment_answers],
+            },
+        )
+    if r.status_code != httpx.codes.CREATED:
+        await log_and_report(
+            text=(
+                f"Chat {user_data.chat_id}: Failed to send assessment "
+                f"({r.status_code=}, {r.content=})"
+            ),
+            logger=logger,
+            level="error",
+            bot=update.get_bot(),
+            parse_mode=None,
+        )
+        return False
```

Comment:
The same question as below about `0` as return value. I think function should raise some exception if it wasn't success.

Reply:
(outdated)